pilot/pkg/model/context.go

```diff
@@ -308,12 +309,36 @@ func (s *StringBool) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// ProxyConfig can only be marshalled using (gogo) jsonpb. However, the rest of node meta is not a proto
+// To allow marshalling, we need to define a custom type that calls out to the gogo marshaller
+type JsonProxyConfig meshconfig.ProxyConfig
```

Comment:
Name is confusing. Suggestions:

`NodeMetaProxyConfig` as it's just a type for embedding/serializing from node.metadata.